app/main.py

```diff
@@ -1,52 +1,37 @@
-import json
-import xml.etree.ElementTree as ET
-
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.book import Book
+from app.display import ConsoleDisplayer, ReverseDisplayer
+from app.prints import PrintConsole, PrintReverse
+from app.serialize import JSONSerializer, XMLSerializer
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            if method_type == "console":
+                displayer = ConsoleDisplayer(book)
+            elif method_type == "reverse":
+                displayer = ReverseDisplayer(book)
+            else:
+                raise ValueError("Unknown display type ...")
+            displayer.display()
+
         elif cmd == "print":
-            book.print_book(method_type)
+            if method_type == "console":
+                prints = PrintConsole(book)
+            elif method_type == "reverse":
+                prints = PrintReverse(book)
+            else:
+                raise ValueError("Unknown print type ...")
+            prints.print()
+
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            if method_type == "json":
+                serializer = JSONSerializer(book)
+            elif method_type == "xml":
+                serializer = XMLSerializer(book)
+            else:
+                raise ValueError("Unknown serialize type")
+            return serializer.serialize()
```

Comment:
Even with such a short functionality of the program, your main function already looks quite massive, think about how you can take type checking out of this function.